### PR TITLE
Merge branch

### PR DESCRIPTION
**Pull request by cgsandford**
Intermediate branch to resolve conflicts between nowcasting feature branch and master, in order to pull the pytest and clize updates into the feature branch.  This pulls clizefication of all the nowcasting CLIs into the feature branch, and leaves the next merge (which should have no conflicts) to be done separately.

Testing:
 - [x] Ran tests and they passed OK

### CODE REVIEW

File: lib/improver/psychrometric_calculations/psychrometric_calculations.py [hunk 1]
```diff
@@ -1037,41 +1023,41 @@ def find_max_in_nbhood_orography(self, orography_cube):
             radius_in_metres).process(orography_cube)
         return max_in_nbhood_orog
 
-    def process(self, temperature, relative_humidity, pressure, orog,
+    def process(self, wet_bulb_temperature, wet_bulb_integral, orog,
```
Comment (benfitzpatrick) [inline lib/improver/psychrometric_calculations/psychrometric_calculations.py]: Codacy Issue found: [Too many local variables (24/20)](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=4756604)

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 12
- Added 1
           

Complexity increasing per file
==============================
- lib/improver/tests/acceptance/test_combine.py  1
- lib/improver/tests/acceptance/test_standardise.py  2
- lib/improver/tests/acceptance/test_between_thresholds.py  1
- lib/improver/tests/acceptance/test_nowcast_extrapolate.py  1
- lib/improver/cli/wet_bulb_temperature_integral.py  1
- lib/improver/tests/acceptance/test_feels_like_temp.py  2
- lib/improver/tests/acceptance/test_sleet_probability.py  1
- lib/improver/tests/acceptance/test_wind_gust_diagnostic.py  1
- lib/improver/tests/acceptance/test_phase_change_level.py  2
- lib/improver/tests/acceptance/test_extract.py  2
- lib/improver/tests/acceptance/test_uv_index.py  2
- lib/improver/tests/acceptance/test_wind_direction.py  1
- lib/improver/tests/psychrometric_calculations/psychrometric_calculations/test_PhaseChangeLevel.py  2
- lib/improver/tests/acceptance/test_convert_to_realizations.py  1
- lib/improver/tests/conftest.py  1
- lib/improver/cli/sleet_probability.py  1
- lib/improver/tests/acceptance/test_recursive_filter.py  1
- lib/improver/calculate_sleet_prob.py  2
- lib/improver/tests/acceptance/test_threshold.py  1
- lib/improver/tests/acceptance/test_nowcast_optical_flow.py  2
- lib/improver/tests/acceptance/test_orographic_enhancement.py  3
- lib/improver/cli/phase_change_level.py  1
- lib/improver/tests/acceptance/test_neighbour_finding.py  2
- lib/improver/tests/acceptance/test_resolve_wind_components.py  1
- lib/improver/tests/acceptance/test_time_lagged_ensembles.py  2
- lib/improver/tests/acceptance/test_create_grid_with_halo.py  1
- lib/improver/tests/acceptance/test_nbhood_iterate_with_mask.py  3
- lib/improver/cli/resolve_wind_components.py  3
- lib/improver/cli/convert_to_realizations.py  4
- lib/improver/tests/acceptance/test_update_grid_metadata.py  2
- lib/improver/tests/acceptance/acceptance.py  9
- lib/improver/metadata/utilities.py  4
- lib/improver/tests/acceptance/test_weighted_blending.py  2
- lib/improver/tests/acceptance/test_extend_radar_mask.py  1
- lib/improver/tests/acceptance/test_percentiles_to_realizations.py  3
- lib/improver/tests/acceptance/test_wet_bulb_temperature.py  2
- lib/improver/tests/acceptance/test_nbhood_land_and_sea.py  3
- lib/improver/tests/acceptance/test_wxcode.py  3
- lib/improver/tests/acceptance/test_wind_downscaling.py  3
- lib/improver/cli/nowcast_accumulate.py  3
- lib/improver/tests/acceptance/test_percentile.py  1
- lib/improver/cli/__init__.py  3
- lib/improver/tests/acceptance/test_blend_adjacent_points.py  2
- lib/improver/tests/acceptance/test_generate_landmask_ancillary.py  1
- lib/improver/tests/acceptance/test_apply_emos_coefficients.py  2
- lib/improver/tests/acceptance/test_spot_extract.py  3
- lib/improver/tests/acceptance/test_nowcast_accumulate.py  1
- lib/improver/tests/acceptance/test_probabilities_to_realizations.py  2
- lib/improver/tests/acceptance/test_estimate_emos_coefficients.py  2
- lib/improver/spotdata/spot_extraction.py  1
- lib/improver/tests/acceptance/test_percentiles_to_probabilities.py  1
- lib/improver/tests/acceptance/test_wet_bulb_temperature_integral.py  1
- lib/improver/tests/acceptance/test_apply_lapse_rate.py  1
- lib/improver/tests/acceptance/test_generate_topography_bands.py  2
- lib/improver/tests/acceptance/test_temporal_interpolate.py  5
- lib/improver/tests/calculate_sleet_prob/test_calculate_sleet_probability.py  2
- lib/improver/tests/acceptance/test_nbhood.py  1
- lib/improver/tests/acceptance/test_temp_lapse_rate.py  2
         

Complexity decreasing per file
==============================
+ lib/improver/cli/nowcast_extrapolate.py  -10
+ lib/improver/tests/utilities/test_GenerateTimeLaggedEnsemble.py  -1
+ lib/improver/wind_calculations/wind_gust_diagnostic.py  -1
+ lib/improver/metadata/forecast_times.py  -8
+ lib/improver/cli/nowcast_optical_flow.py  -4
         

Clones added
============
- lib/improver/tests/acceptance/test_standardise.py  2
- lib/improver/tests/acceptance/test_wind_direction.py  1
- lib/improver/tests/psychrometric_calculations/psychrometric_calculations/test_PhaseChangeLevel.py  13
- lib/improver/tests/acceptance/test_nowcast_optical_flow.py  2
- lib/improver/tests/acceptance/test_orographic_enhancement.py  2
- lib/improver/tests/acceptance/test_nbhood_iterate_with_mask.py  1
- lib/improver/tests/acceptance/test_wxcode.py  1
- lib/improver/tests/acceptance/test_wind_downscaling.py  5
- lib/improver/tests/acceptance/test_percentile.py  2
- lib/improver/tests/acceptance/test_apply_emos_coefficients.py  2
- lib/improver/tests/acceptance/test_estimate_emos_coefficients.py  2
         

Clones removed
==============
+ lib/improver/tests/utilities/test_GenerateTimeLaggedEnsemble.py  -5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=4756604&bid=15863195)

**Comment by cgsandford** [comment]
This seems huge but from a quick skim is as expected; vast majority of changes attributable to removal of all bats tests and replacement with pytests implementation.

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 12
- Added 1
           

Complexity increasing per file
==============================
- lib/improver/tests/acceptance/test_wind_direction.py  1
- lib/improver/tests/acceptance/test_spot_extract.py  3
- lib/improver/tests/acceptance/test_create_grid_with_halo.py  1
- lib/improver/tests/acceptance/test_nowcast_accumulate.py  1
- lib/improver/tests/acceptance/test_wind_gust_diagnostic.py  1
- lib/improver/tests/acceptance/test_percentiles_to_probabilities.py  1
- lib/improver/cli/sleet_probability.py  1
- lib/improver/tests/acceptance/test_wet_bulb_temperature.py  2
- lib/improver/tests/acceptance/test_threshold.py  1
- lib/improver/tests/acceptance/test_nbhood_land_and_sea.py  3
- lib/improver/tests/acceptance/test_wet_bulb_temperature_integral.py  1
- lib/improver/tests/acceptance/test_phase_change_level.py  2
- lib/improver/tests/acceptance/acceptance.py  9
- lib/improver/tests/acceptance/test_generate_topography_bands.py  2
- lib/improver/cli/phase_change_level.py  1
- lib/improver/tests/acceptance/test_nbhood_iterate_with_mask.py  3
- lib/improver/cli/resolve_wind_components.py  3
- lib/improver/calculate_sleet_prob.py  2
- lib/improver/cli/wet_bulb_temperature_integral.py  1
- lib/improver/tests/acceptance/test_combine.py  1
- lib/improver/tests/acceptance/test_neighbour_finding.py  2
- lib/improver/tests/acceptance/test_update_grid_metadata.py  2
- lib/improver/tests/acceptance/test_apply_emos_coefficients.py  2
- lib/improver/cli/__init__.py  3
- lib/improver/tests/calculate_sleet_prob/test_calculate_sleet_probability.py  2
- lib/improver/tests/acceptance/test_estimate_emos_coefficients.py  2
- lib/improver/tests/acceptance/test_uv_index.py  2
- lib/improver/tests/acceptance/test_wind_downscaling.py  3
- lib/improver/cli/nowcast_accumulate.py  3
- lib/improver/tests/acceptance/test_standardise.py  2
- lib/improver/tests/acceptance/test_extend_radar_mask.py  1
- lib/improver/tests/acceptance/test_sleet_probability.py  1
- lib/improver/metadata/utilities.py  4
- lib/improver/tests/acceptance/test_weighted_blending.py  2
- lib/improver/tests/acceptance/test_temp_lapse_rate.py  2
- lib/improver/tests/acceptance/test_probabilities_to_realizations.py  2
- lib/improver/cli/convert_to_realizations.py  4
- lib/improver/tests/psychrometric_calculations/psychrometric_calculations/test_PhaseChangeLevel.py  2
- lib/improver/tests/acceptance/test_between_thresholds.py  1
- lib/improver/tests/acceptance/test_wxcode.py  3
- lib/improver/tests/acceptance/test_temporal_interpolate.py  5
- lib/improver/tests/acceptance/test_nbhood.py  1
- lib/improver/tests/acceptance/test_resolve_wind_components.py  1
- lib/improver/tests/acceptance/test_percentiles_to_realizations.py  3
- lib/improver/tests/acceptance/test_generate_landmask_ancillary.py  1
- lib/improver/tests/acceptance/test_nowcast_extrapolate.py  1
- lib/improver/tests/acceptance/test_percentile.py  1
- lib/improver/tests/conftest.py  1
- lib/improver/tests/acceptance/test_convert_to_realizations.py  1
- lib/improver/tests/acceptance/test_recursive_filter.py  1
- lib/improver/tests/acceptance/test_blend_adjacent_points.py  2
- lib/improver/spotdata/spot_extraction.py  1
- lib/improver/tests/acceptance/test_feels_like_temp.py  2
- lib/improver/tests/acceptance/test_extract.py  2
- lib/improver/tests/acceptance/test_orographic_enhancement.py  3
- lib/improver/tests/acceptance/test_time_lagged_ensembles.py  2
- lib/improver/tests/acceptance/test_nowcast_optical_flow.py  2
- lib/improver/tests/acceptance/test_apply_lapse_rate.py  1
         

Complexity decreasing per file
==============================
+ lib/improver/cli/nowcast_optical_flow.py  -4
+ lib/improver/cli/nowcast_extrapolate.py  -10
+ lib/improver/metadata/forecast_times.py  -8
+ lib/improver/tests/utilities/test_GenerateTimeLaggedEnsemble.py  -1
+ lib/improver/wind_calculations/wind_gust_diagnostic.py  -1
         

Clones added
============
- lib/improver/tests/acceptance/test_wind_direction.py  1
- lib/improver/tests/acceptance/test_nbhood_iterate_with_mask.py  1
- lib/improver/tests/acceptance/test_apply_emos_coefficients.py  2
- lib/improver/tests/acceptance/test_estimate_emos_coefficients.py  2
- lib/improver/tests/acceptance/test_wind_downscaling.py  5
- lib/improver/tests/acceptance/test_standardise.py  2
- lib/improver/tests/psychrometric_calculations/psychrometric_calculations/test_PhaseChangeLevel.py  13
- lib/improver/tests/acceptance/test_wxcode.py  1
- lib/improver/tests/acceptance/test_percentile.py  2
- lib/improver/tests/acceptance/test_orographic_enhancement.py  2
- lib/improver/tests/acceptance/test_nowcast_optical_flow.py  2
         

Clones removed
==============
+ lib/improver/tests/utilities/test_GenerateTimeLaggedEnsemble.py  -5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=4756604&bid=15863195)

File: lib/improver/psychrometric_calculations/psychrometric_calculations.py (hunk 1 above)
Comment (benfitzpatrick) [inline lib/improver/psychrometric_calculations/psychrometric_calculations.py]: Codacy Issue found: [Too many local variables (24/20)](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=4756604)

**Pull request merged**